Source/VMware.vSphereDSC/Tests/Unit/VMHostNetwork/VMHostVirtualPortGroup.Unit.Tests.ps1

```diff
@@ -0,0 +1,492 @@
+<#
+Copyright (c) 2018 VMware, Inc.  All rights reserved
+
+The BSD-2 license (the "License") set forth below applies to all parts of the Desired State Configuration Resources for VMware project.  You may not use this file except in compliance with the License.
+
+BSD-2 License
+
+Redistribution and use in source and binary forms, with or without modification, are permitted provided that the following conditions are met:
+
+Redistributions of source code must retain the above copyright notice, this list of conditions and the following disclaimer.
+
+Redistributions in binary form must reproduce the above copyright notice, this list of conditions and the following disclaimer in the documentation and/or other materials provided with the distribution.
+
+THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+#>
+
+using module VMware.vSphereDSC
+
+$script:moduleName = 'VMware.vSphereDSC'
+
+InModuleScope -ModuleName $script:moduleName {
+    try {
+        $unitTestsFolder = Join-Path (Join-Path (Get-Module VMware.vSphereDSC -ListAvailable).ModuleBase 'Tests') 'Unit'
+        $modulePath = $env:PSModulePath
+        $resourceName = 'VMHostVirtualPortGroup'
+
+        . "$unitTestsFolder\TestHelpers\TestUtils.ps1"
+
+        # Calls the function to Import the mocked VMware.VimAutomation.Core module before all tests.
+        Invoke-TestSetup
+
+        . "$unitTestsFolder\TestHelpers\Mocks\MockData.ps1"
+        . "$unitTestsFolder\TestHelpers\Mocks\VMHostVirtualPortGroupMocks.ps1"
+
+        Describe 'VMHostVirtualPortGroup\Set' -Tag 'Set' {
```

Comment:
It's good idea to have negative tests when New/Remove/Set-VirtualPortGroup fail throws (terminating)  and writes(non-terminating) error. You have tests when those cmdlets fail in case of remove when not port group and add when there is but I think it is good to check the behavior of the resources in case of non-terminating error to cover try {-ErrorAction Stop} catch {}

BTW I'm not sure all PowerCLI cmdlet handle -ErrorAction parameter properly I hope they do but not sure :)

Reply:
New Tests covering this particular case will be added in a separate PR.